src/coordination/coordinator_cluster_state.cpp

```diff
@@ -0,0 +1,124 @@
+// Copyright 2024 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#ifdef MG_ENTERPRISE
+
+#include "nuraft/coordinator_cluster_state.hpp"
+
+namespace memgraph::coordination {
+
+auto CoordinatorClusterState::MainExists() const -> bool {
+  return std::any_of(instance_roles.begin(), instance_roles.end(), [](auto const &entry) {
+    return entry.second == replication_coordination_glue::ReplicationRole::MAIN;
+  });
+}
+
+auto CoordinatorClusterState::IsMain(std::string const &instance_name) const -> bool {
+  auto const it = instance_roles.find(instance_name);
+  return it != instance_roles.end() && it->second == replication_coordination_glue::ReplicationRole::MAIN;
```

Comment:
This is a cpp file consider `using memgraph::replication_coordination_glue::ReplicationRole;` to reduce verbosity throughout the file.